homeassistant/components/sma/entities.py

```diff
@@ -0,0 +1,776 @@
+        key="grid_power_factor_excitation",
+        name="Grid Power Factor Excitation",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER_FACTOR,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    "current_l1": SensorEntityDescription(
+        key="current_l1",
+        name="Current L1",
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.CURRENT,
+        entity_registry_enabled_default=False,
+    ),
+    "current_l2": SensorEntityDescription(
+        key="current_l2",
+        name="Current L2",
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.CURRENT,
+        entity_registry_enabled_default=False,
+    ),
+    "current_l3": SensorEntityDescription(
+        key="current_l3",
+        name="Current L3",
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.CURRENT,
+        entity_registry_enabled_default=False,
+    ),
+    "current_total": SensorEntityDescription(
+        key="current_total",
+        name="Current Total",
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.CURRENT,
+    ),
+    "voltage_l1": SensorEntityDescription(
+        key="voltage_l1",
+        name="Voltage L1",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        entity_registry_enabled_default=False,
+    ),
+    "voltage_l2": SensorEntityDescription(
+        key="voltage_l2",
+        name="Voltage L2",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        entity_registry_enabled_default=False,
+    ),
+    "voltage_l3": SensorEntityDescription(
+        key="voltage_l3",
+        name="Voltage L3",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        entity_registry_enabled_default=False,
+    ),
+    "total_yield": SensorEntityDescription(
+        key="total_yield",
+        name="Total Yield",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+    ),
+    "daily_yield": SensorEntityDescription(
+        key="daily_yield",
+        name="Daily Yield",
+        native_unit_of_measurement=ENERGY_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
```

Comment:
There's no state class here. What kind of measurement is this? Maybe use state class total and reset the sensor every day via `last_reset` attribute? It's ok to leave the state class out if we don't have all the information.

Reply:
This value will go back to 0 on a daily basis. So `SensorStateClass.TOTAL` would be applicable. Do we need to calculate `last_reset` our self or will this be handles automatically?